### PR TITLE
Fixes #468: markReadableIfBuilt does not work properly

### LINKED ISSUE #468: markReadableIfBuilt does not work properly

### PR DESCRIPTION
**Pull request by nblintao**

### CODE REVIEW

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java [hunk 1]
```diff
@@ -2398,4 +2399,31 @@ public void closeWhileBuilding() throws Exception {
         int count3 = timer.getCount(FDBStoreTimer.Events.COMMIT);
         assertThat("No more commits should have occurred", count3, Matchers.is(count2));
     }
+
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java]: Extra blank line

File: docs/ReleaseNotes.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ In order to simplify typed record stores, the `FDBRecordStoreBase` class was tur
 ### NEXT_RELEASE
 
 * **Bug fix** `OnlineIndexer` fails to `getPrimaryKeyBoundaries` when there is no boundary [(Issue #460)](https://github.com/FoundationDB/fdb-record-layer/issues/460)
-* **Bug fix** Fix 2 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
+* **Bug fix** markReadableIfBuilt does not work properly [(Issue #468)](https://github.com/FoundationDB/fdb-record-layer/issues/468)
```
Comment (alecgrieser) [inline docs/ReleaseNotes.md]:
```suggestion
* **Bug fix** The `markReadableIfBuilt` method of `OnlineIndexer` now waits for the index to be marked readable before returning [(Issue #468)](https://github.com/FoundationDB/fdb-record-layer/issues/468)
```

Or something along those lines.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java [hunk 3]
```diff
@@ -1007,9 +1008,11 @@ public void buildIndex() {
     /**
      * Wait for asynchronous index build to complete.
      * @param buildIndexFuture the result of {@link #buildIndexAsync}
+     * @param <T> return type of function to run
+     * @return future that will contain the result of {@code retriable} after successful run and commit
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java]:
```suggestion
     * @return future that will contain the result of {@code buildIndexFuture} after successful run and commit
```

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java [hunk 4]
```diff
@@ -1007,9 +1008,11 @@ public void buildIndex() {
...
      */
-    public void asyncToSync(@Nonnull CompletableFuture<Void> buildIndexFuture) {
-        runner.asyncToSync(FDBStoreTimer.Waits.WAIT_ONLINE_BUILD_INDEX, buildIndexFuture);
+    public <T> T asyncToSync(@Nonnull CompletableFuture<T> buildIndexFuture) {
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java]: It's weird to me that this is a public method, but this is probably not the right PR to deal with that.

Comment (alecgrieser) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java]: Is this being modified to make the unit test easier?

Comment (nblintao) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java]: - Yes, I think it make sense not exposing `asyncToSync` but it is a breaking change.

- Yes, it was modified to make the unit test easier.

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java (hunk 1 above)
Comment (nblintao) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerTest.java]: Done.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java (hunk 4 above)
Comment (alecgrieser) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java]: I think we could probably mark it either `INTERNAL` or `UNSTABLE` and then think about removing it (i.e., making it...package private?) in a future release.

**Pull request merged**